solr/core/src/test/org/apache/solr/cloud/OverseerModifyCollectionTest.java

```diff
@@ -42,29 +39,27 @@ public static void setupCluster() throws Exception {
   }
 
   @Test
+  @SuppressWarnings({"unchecked"})
   public void testModifyColl() throws Exception {
 
     final String collName = "modifyColl";
 
     CollectionAdminRequest.createCollection(collName, "conf1", 1, 2)
         .process(cluster.getSolrClient());
 
-    // TODO create a modifyCollection() method on CollectionAdminRequest
-    ModifiableSolrParams p1 = new ModifiableSolrParams();
-    p1.add("collection", collName);
-    p1.add("action", "MODIFYCOLLECTION");
-    p1.add("collection.configName", "conf2");
-    cluster.getSolrClient().request(new GenericSolrRequest(POST, COLLECTIONS_HANDLER_PATH, p1));
+    //Modify configSet
+    RequestStatusState requestStatusState = CollectionAdminRequest.modifyCollection(collName,
+            map("collection.configName", "conf2")
+    ).processAndWait(cluster.getSolrClient(), DEFAULT_TIMEOUT);
+    assertEquals(requestStatusState.getKey(), "completed");
```

Comment:
RequestStatusState is an enum, and is thus immutable.  Why not just do an instance check?

Reply:
And I presume the changes here fail without the underlying fix?

Reply:
Yes, it does fail without this fix